packages/@aws-cdk/aws-sagemaker-alpha/lib/model-data.ts

```diff
@@ -6,7 +6,58 @@ import { hashcode } from './private/util';
 
 // The only supported extension for local asset model data
 // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-model-containerdefinition.html#cfn-sagemaker-model-containerdefinition-modeldataurl
-const ARTIFACT_EXTENSION = '.tar.gz';
+const COMPRESSED_ARTIFACT_EXTENSION = '.tar.gz';
+
+/**
+ * Specifies how the ML model data is prepared.
+ */
+export enum CompressionType {
+  /**
+   * If you choose `CompressionType.GZIP` and choose `S3DataType.S3_OBJECT` as the value of `s3DataType`,
+   * S3 URI identifies an object that is a gzip-compressed TAR archive.
+   * SageMaker will attempt to decompress and untar the object during model deployment.
+   */
+  GZIP = 'Gzip',
+  /**
+   * If you choose `CompressionType.NONE` and choose `S3DataType.S3_PREFIX` as the value of `s3DataType`,
+   * S3 URI identifies a key name prefix, under which all objects represents the uncompressed ML model to deploy.
+   *
+   * If you choose `CompressionType.NONE`, then SageMaker will follow rules below when creating model data files
+   * under `/opt/ml/model` directory for use by your inference code:
```

Comment:
The `CompressionType` and `S3DataType` enums seem to be used a lot in conjunction with each other, and when their behaviour changes this drastically between combinations, it makes me wonder if there is any way we can combine the two? Right now, for instance, there doesn't seem to be anything documented about what will happen if I combine `CompressionType.GZIP` and `S3DataType.S3_PREFIX`, if that's a valid combination.